subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/DefaultHttpSettings.java

```diff
@@ -16,24 +16,37 @@
 package org.gradle.internal.resource.transport.http;
 
 
+import org.apache.http.conn.ssl.DefaultHostnameVerifier;
 import org.gradle.authentication.Authentication;
 
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSession;
 import java.util.Collection;
 
 public class DefaultHttpSettings implements HttpSettings {
     private final Collection<Authentication> authenticationSettings;
     private final SslContextFactory sslContextFactory;
+    private final HostnameVerifier hostnameVerifier;
     private HttpProxySettings proxySettings;
     private HttpProxySettings secureProxySettings;
     private HttpTimeoutSettings timeoutSettings;
 
+    public static DefaultHttpSettings allowUntrustedSslConnections(Collection<Authentication> authenticationSettings) {
+        return new DefaultHttpSettings(authenticationSettings, new AllTrustingSslContextFactory(), ALL_TRUSTING_HOSTNAME_VERIFIER);
```

Comment:
Can `AllTrustingSslContextFactory` be a similar constant?

Reply:
I removed the class and made it into a constant.